File: src/main/java/com/hirrao/xnntech/main/CommonProxy.java
package com.hirrao.xnntech.main;

import com.hirrao.xnntech.loader.ConfigLoader;
import com.hirrao.xnntech.loader.MetaTileEntitiesLoader;
import com.hirrao.xnntech.loader.RecipesLoader;

import cpw.mods.fml.common.event.*;

@SuppressWarnings("unused")
public class CommonProxy {

    // preInit "Run before anything else. Read your config, create blocks, items, etc, and register them with the
    // GameRegistry." (Remove if not needed)
    public void preInit(FMLPreInitializationEvent event) {
        ConfigLoader.load();
        MetaTileEntitiesLoader.load();
    }

    // load "Do your mod setup. Build whatever data structures you care about. Register recipes." (Remove if not needed)
    public void init(FMLInitializationEvent event) {
        RecipesLoader.load();
    }

    // postInit "Handle interaction with other mods, complete your setup based on this." (Remove if not needed)
    public void postInit(FMLPostInitializationEvent event) {}

    public void CompleteInit(FMLLoadCompleteEvent event) {

    }

    // register server commands in this event handler (Remove if not needed)
    public void serverStarting(FMLServerStartingEvent event) {}
}

File: src/main/java/com/hirrao/xnntech/main/CommonProxy.kt
package com.hirrao.xnntech.main

import com.hirrao.xnntech.loader.loadConfig
import com.hirrao.xnntech.loader.loadMetaTileEntities
import com.hirrao.xnntech.loader.loadRecipes
import cpw.mods.fml.common.event.*

@Suppress("unused")
open class CommonProxy {
    // preInit "Run before anything else. Read your config, create blocks, items, etc, and register them with the
    // GameRegistry." (Remove if not needed)
    fun preInit(event: FMLPreInitializationEvent) {
        loadConfig()
        loadMetaTileEntities()
    }

    // load "Do your mod setup. Build whatever data structures you care about. Register recipes." (Remove if not needed)
    fun init(event: FMLInitializationEvent) {
        loadRecipes()
    }

    // postInit "Handle interaction with other mods, complete your setup based on this." (Remove if not needed)
    fun postInit(event: FMLPostInitializationEvent) {}

    fun completeInit(event: FMLLoadCompleteEvent) {
    }

    // register server commands in this event handler (Remove if not needed)
    fun serverStarting(event: FMLServerStartingEvent) {}
}
